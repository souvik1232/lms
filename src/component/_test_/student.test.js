import React from 'react';
import { shallow } from 'enzyme';
import Student from '../student/student'


describe('<Student /> with no props', () => {
    const container = shallow(<Student />);
    it('should match the snapshot', () => {
        expect(container.html()).toMatchSnapshot();
    });
});

describe('Login Component', () => {
    it('should render without throwing an error', () => {
        expect(shallow(<Student />).exists()).toBe(true)
    })
});
describe("Button Component Tests", () => {
    it("Renders correctly in DOM", () => {
        shallow(
            <button text="Test" />
        );
    });
    it("Expects to find button HTML element in the DOM", () => {
        const wrapper = shallow(<button text="test" />)
        expect(wrapper.find('button')).toHaveLength(1);
    });
    it("Expects to find button HTML element with className pgn in the DOM", () => {
        const wrapper = shallow(<button className="pgn" text="test" />)
        expect(wrapper.find('button.pgn')).toHaveLength(1);
    });
    it("Expects to run onClick function when button is pressed in the DOM", () => {
        const mockCallBackClick = jest.fn();
        const wrapper = shallow(<button onClick={mockCallBackClick} className="btn" text="test" />);
        wrapper.find('button').simulate('click');
        expect(mockCallBackClick.mock.calls.length).toEqual(1);
    });
})
describe('counter testing ', () => {
    test('render the title ', () => {
        const wrapper = shallow(<Student />);
        expect(wrapper.find('.t1').text()).toContain("STUDENT DETAILS");
    })
})

describe('prop testing',()=>{
    const data=[{
        id: 5,
        student: "Student Student",
        student_id: 9,
        sid: "SI-1001",
        alt_number: null,
        relation_with_alt_number_holder: null,
        current_location: null,
        current_address: null,
        git_link: null,
        year_of_experience: null
      }];
    it('Renders props',()=>{
        const mountt = shallow(
            <Student data={data}/>
        )
        console.log(mountt.debug());
        const result = mountt.find('.Tabhdr'); 
        console.log("Result debug", result.debug()); 
        
    })
    // const wrapper = shallowExample({ item: valueProps })
    // // console.log("wrapper details", wrapper.debug());
    // const acronym=wrapper.find('.componentName');
    // expect(acronym.text()).toBe(valueProps.acronym)
    
})