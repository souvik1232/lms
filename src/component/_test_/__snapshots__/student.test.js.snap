// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mentor /> with no props should match the snapshot 1`] = `"<div class=\\"detailscontainer12\\"><div class=\\"details2\\"><div class=\\"lac\\"><div class=\\"t1\\">STUDENT DETAILS</div><div class=\\"t1-holder\\"><img class=\\"upload\\" alt=\\"\\"/><button type=\\"button\\" class=\\"buto1 btn btn-primary\\">Add Student</button></div></div><br/><br/><div class=\\"tbl-cnt\\"><div class=\\"MuiTableContainer-root tcont\\"><table class=\\"MuiTable-root main-table\\" aria-label=\\"customized table\\"><thead class=\\"MuiTableHead-root t-head\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Student ID</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Name</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Email ID</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Mobile No.</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Course</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Mentor</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Score</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Week</th></tr></thead><tbody class=\\"MuiTableBody-root t-body\\"></tbody></table></div></div></div></div>"`;

exports[`<Student /> with no props should match the snapshot 1`] = `"<div class=\\"detailscontainer12\\"><div class=\\"details2\\"><div class=\\"lac\\"><div class=\\"t1\\">STUDENT DETAILS</div><div class=\\"t1-holder\\"><img class=\\"upload\\" alt=\\"\\"/><button type=\\"button\\" class=\\"buto1 btn btn-primary\\">Add Student</button></div></div><br/><br/><div class=\\"tbl-cnt\\"><div class=\\"MuiTableContainer-root tcont\\"><table class=\\"MuiTable-root main-table\\" aria-label=\\"customized table\\"><thead class=\\"MuiTableHead-root t-head\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Student ID</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Name</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Email ID</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Mobile No.</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Course</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Mentor</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Score</th><th class=\\"MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter\\" scope=\\"col\\" id=\\"th\\">Week</th></tr></thead><tbody class=\\"MuiTableBody-root t-body\\"></tbody></table></div></div></div></div>"`;
